src/testing/testing-utils.ts

```diff
@@ -95,3 +95,99 @@ function getAppUrl(config: d.Config, browserUrl: string, appFileName: string) {
 
   return browserUrl;
 }
+
+/**
+ * Utility for silencing `console` functions in tests.
+ *
+ * When this function is first called it grabs a reference to the `log`,
+ * `error`, and `warn` functions on `console` and then returns a per-test setup
+ * function which sets up a fresh set of mocks (via `jest.fn()`) and then
+ * assigns them to each of these functions. This setup function will return a
+ * reference to each of the three mock functions so tests can make assertions
+ * about their calls and so on.
+ *
+ * Because references to the original `.log`, `.error`, and `.warn` functions
+ * exist in closure within the function, it can use an `afterAll` call to clean
+ * up after itself and ensure that the original implementations are restored
+ * after the test suite finishes.
+ *
+ * An example of using this to silence log statements in a single test could look
+ * like this:
+ *
+ * ```ts
+ * describe("my-test-suite", () => {
+ *   const setupConsoleMocks = setupConsoleMocker()
+ *
+ *   it("should log a message", () => {
+ *     const { logMock } = setupConsoleMocks();
+ *     myFunctionWhichLogs(foo, bar);
+ *     expect(logMock).toBeCalledWith('my log message');
+ *   })
+ * })
+ * ```
+ *
+ * @returns a per-test mock setup function
+ */
+export function setupConsoleMocker(): ConsoleMocker {
+  const originalLog = console.log;
+  const originalWarn = console.warn;
+  const originalError = console.error;
+
+  afterAll(() => {
+    console.log = originalLog;
+    console.warn = originalWarn;
+    console.error = originalError;
+  });
+
+  function setupConsoleMocks() {
+    const logMock = jest.fn();
+    const warnMock = jest.fn();
+    const errorMock = jest.fn();
+
+    console.log = logMock;
+    console.warn = warnMock;
+    console.error = errorMock;
+
+    return {
+      logMock,
+      warnMock,
+      errorMock,
+    };
+  }
+  return setupConsoleMocks;
+}
+
+interface ConsoleMocker {
+  (): {
+    logMock: jest.Mock<typeof console.log>;
+    warnMock: jest.Mock<typeof console.warn>;
+    errorMock: jest.Mock<typeof console.error>;
+  };
+}
+
+// export function withSilentWarn<T extends () => Promise<any>>(
```

Comment:
Do we need this still? If not, can we remove it?

Reply:
oops 🤦‍♀️